### PR TITLE
Add mempool api to devenv

### PR DESCRIPTION
**Pull request by AshtonStephens**
## Summary of Changes

Fixes issue: https://github.com/stacks-network/sbtc/issues/196

I added three services to the docker compose file at `devenv` root that are all needed to support the mempool API needed by the current release of Leather wallet.

I've also copy pasted the `docker` directory from the original [mempool github](https://github.com/mempool/mempool/tree/master/docker) into the `devenv` folder and opened the ticket https://github.com/stacks-network/sbtc/issues/232 to make the `mempool` subdirectory of the devenv folder match the other containers.

### Risks

1. This setup could not work and make people believe their code is okay.
2. The subdirectory that was copy pasted will likely be confusing unless we deal with it.
3. The number of docker containers could make this non-runnable by hackathon participants. The number of containers needed to run devenv is getting out of control.

### How were these changes tested?

This was tested locally by running the following at project root

```bash
cd devenv && sudo ./up.sh
```

After that finishes, you should be able to put the following url into your browser to get mempool api from port 8083. For example, the url `http://127.0.0.1:8083/api/v1/fees/recommended` should return something like 

```json
{"fastestFee":1,"halfHourFee":1,"hourFee":1,"economyFee":1,"minimumFee":1}
```

### What future testing should occur?

Further local testing should be done by making a demo sBTC project with the leather wallet to ensure that this works correctly on devenv.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [x] My changes generate no new warnings
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: devenv/docker-compose.yml [hunk 1]
```diff
@@ -219,3 +219,62 @@ services:
...
+      DATABASE_HOST: "mempool-db"
+      DATABASE_DATABASE: "mempool"
+      DATABASE_USERNAME: "mempool"
+      DATABASE_PASSWORD: "mempool"
+      STATISTICS_ENABLED: "true"
+    volumes:
+      - $PWD/mempool/data:/backend/cache
+    command: "./wait-for-it.sh mempool-db:3306 --timeout=720 --strict -- ./start.sh"
+
+  mempool-db:
```
[the remark this answers is not in the document]
Comment (AshtonStephens) [reply devenv/docker-compose.yml:269]: Done.

File: devenv/mempool/backend/Dockerfile
[the remark this answers is not in the document]
Comment (AshtonStephens) [reply devenv/mempool/backend/Dockerfile:1]: I ran this on my mac, so I think we should be good

File: devenv/mempool/backend/mempool-config.json
[the remark this answers is not in the document]
Comment (AshtonStephens) [reply devenv/mempool/backend/mempool-config.json:1]: I'm wondering if we even need to have the `devenv/mempool` directory in the project at all. If someone wants to run this on its own they should use the actual mempool repo. I can include a readme that directs there instead.

File: devenv/docker-compose.yml [hunk 2]
```diff
@@ -219,3 +219,62 @@ services:
...
+    container_name: mempool-db
+    user: "1000:1000"
+    restart: on-failure
+    environment:
+      MYSQL_DATABASE: "mempool"
+      MYSQL_USER: "mempool"
+      MYSQL_PASSWORD: "mempool"
+      MYSQL_ROOT_PASSWORD: "admin"
+    volumes:
+      - $PWD/mempool/mysql/data:/var/lib/mysql
```
[the remark this answers is not in the document]
Comment (AshtonStephens) [reply devenv/docker-compose.yml:280]: The `context` only matters if you're building it I believe. I believe the $PWD here is only specifying where to dump the logs.

**Comment by AshtonStephens** [comment]
In the end, there were some issues around folder access for mariadb. https://github.com/MariaDB/mariadb-docker/issues/256

My big mistake was not including the empty directory with the correct permissions within this PR. They were already there for me locally so I didn't catch it 😮‍💨 Glad you ran this Tyler

**Comment by AshtonStephens** [comment]
Put out another review commit to review - I'd appreciate if someone else tested it as it has a history of working on only my computers. I can confirm that it works fine on both my ARM and my x86 computer.

**Comment by friedger** [comment]
build works, but when using up, it complains about port in use 3306 on ubuntu. Looks like I am running mysql/mariadbd already

**Comment by AshtonStephens** [comment]
> build works, but when using up, it complains about port in use 3306 on ubuntu. Looks like I am running mysql/mariadbd already

@friedger is that typical or can we assume that most developers won't already be running the server? I think port collisions could be a problem for all these services.

**Comment by friedger** [comment]
Currently, I don't know why it is running.. Lol

**Review by friedger — APPROVED** [review]
Works for me on ubuntu after I stopped mysql

**Comment by CAGS295** [comment]
> The number of docker containers could make this non-runnable by hackathon participants. The number of containers needed to run devenv is getting out of control.

Devenv is going from "full node" to "full ecosystem" lol.

**Comment by friedger** [comment]
Probably, hackathon participants won't bring their stations @AshtonStephens

**Pull request merged**